### ISSUE TITLE
WebSocket connection closed Error: WebSocket connection closed

### ISSUE THREAD

**Issue by szymon-niedrygas** [cypress-io/cypress#29033]
### Current behavior

After a few or a dozen or so minutes, I get the following error locally:
**WebSocket connection closed
Error: WebSocket connection closed**
Remotely on build agents from TeamCity, the suite simply stops and gets a timeout.
This situation occurs for Chrome and Chromium browsers. Somehow those browsers hangs. For Elektron it is ok.
Cypress Version 13.6.6 and several previous ones

### Desired behavior

Test run can be finished. 

### Test code to reproduce

Run suite that have around 120 tests

### Cypress Version

13.6.6

### Node version

18.17.0

### Operating System

Windows 11 Pro 

### Debug Logs

```shell
`cypress:lifecycle:ProjectConfigManager promise resolved for id 'inv11085' with value [
  'XXX.cy.js.mp4'
] +13ms
  cypress:server:run No video found after spec ran - skipping compression. Video path: C:XXX.cy.js.mp4 +16ms
  cypress:server:run attempting to close the browser tab +5ms
  cypress:server:browsers:browser-cri-client Target.attachedToTarget { targetId: 'EEC464CED6509D700534AEC6A27E6D13', type: 'page', title: '', url: 'about:blank', attached: true, canAccessOpener: false, browserContextId: '529B59B0F277052F102A76B25275EC01' } +5s
  cypress:server:browsers:browser-cri-client Not an extra target (id: EEC464CED6509D700534AEC6A27E6D13) +1ms
  cypress:server:browsers:browser-cri-client currently attached targets EDF12D11DE9C738938E42DD3D68B91B5 false +62ms
  cypress:server:browsers:browser-cri-client closing current target EDF12D11DE9C738938E42DD3D68B91B5 +0ms
  cypress:server:browsers:browser-cri-client target closed EDF12D11DE9C738938E42DD3D68B91B5 +42ms
  cypress:server:browsers:cri-client closing cri client { closed: false, target: 'EDF12D11DE9C738938E42DD3D68B91B5' } +4s
  cypress:server:browsers:cri-client closed cri client { closed: true, target: 'EDF12D11DE9C738938E42DD3D68B91B5' } +2ms
  cypress:server:browsers:browser-cri-client target client closed EDF12D11DE9C738938E42DD3D68B91B5 +2ms
  cypress:server:browsers:cri-client connecting { connected: false, target: 'EEC464CED6509D700534AEC6A27E6D13' } +0ms
WebSocket connection closed
Error: WebSocket connection closed
    at _._handleConnectionClose (<embedded>:2466:579841)
    at A.<anonymous> (<embedded>:2466:577905)
    at Object.onceWrapper (node:events:629:26)
    at A.emit (node:events:514:28)
    at A.emitClose (<embedded>:2466:558129)
    at Socket.W (<embedded>:2466:565440)
    at Socket.emit (node:events:514:28)
    at TCP.<anonymous> (node:net:323:12)
  cypress:server:cloud:api request to url: POST https://api.cypress.io/exceptions with params: {"body":{"err":{"name":"Error","message":"WebSocket connection closed","stack":"Error: WebSocket connection closed\n    at _._handleConnectionClose (<embedded>:2466:579841)\n    at A.<anonymous> (<embedded>:2466:577905)\n    at Object.onceWrapper (node:events:629:26)\n    at A.emit (node:events:514:28)\n    at A.emitClose (<embedded>:2466:558129)\n    at Socket.W (<embedded>:2466:565440)\n    at Socket.emit (node:events:514:28)\n    at TCP.<anonymous> (node:net:323:12)"},"version":"13.6.6","osName":"win32","osVersion":"10.0.22631","osCpus":[{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":11378984,"nice":0,"sys":13391484,"idle":583304468,"irq":1269546}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":8499531,"nice":0,"sys":6671953,"idle":592903062,"irq":524062}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":17181484,"nice":0,"sys":16562921,"idle":574330140,"irq":351968}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":9343781,"nice":0,"sys":8180281,"idle":590550484,"irq":91578}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":12500734,"nice":0,"sys":12319859,"idle":583253953,"irq":154593}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10524265,"nice":0,"sys":9853312,"idle":587696953,"irq":89593}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":12161921,"nice":0,"sys":19248312,"idle":576664296,"irq":95218}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10983187,"nice":0,"sys":8488859,"idle":588602484,"irq":62187}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":11548156,"nice":0,"sys":15596375,"idle":580930015,"irq":197968}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10455437,"nice":0,"sys":7677515,"idle":589941578,"irq":166453}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":11157703,"nice":0,"sys":8914812,"idle":588002015,"irq":122234}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10676250,"nice":0,"sys":5484953,"idle":591913328,"irq":95390}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10900187,"nice":0,"sys":8007906,"idle":589166437,"irq":106734}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":10992093,"nice":0,"sys":4717609,"idle":592364828,"irq":105156}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":12558500,"nice":0,"sys":6958015,"idle":588558000,"irq":103640}},{"model":"Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz","speed":2400,"times":{"user":11461656,"nice":0,"sys":5684921,"idle":590927953,"irq":68156}}],"osMemory":{"free":13364658176,"total":34073567232}},"headers":{"x-os-name":"win32","x-cypress-version":"13.6.6"}} and token: undefined +0ms
  cypress:network:agent addRequest called { isHttps: true, href: 'https://api.cypress.io/exceptions' } +5s
  cypress:network:connect beginning getAddress { hostname: 'api.cypress.io', port: 443 } +3s
  cypress:server:browsers killing browser process +5s
  cypress:server:browsers:chrome closing remote interface client { options: { gracefulShutdown: true } } +4s
  cypress:server:browsers:cri-client not closing, cri client is already closed { closed: true, target: 'EDF12D11DE9C738938E42DD3D68B91B5' } +64ms
  cypress:server:browsers:chrome closing chrome +1ms
  cypress:proxy:http:util:prerequests metrics: { browserPreRequestsReceived: 15673, proxyRequestsReceived: 10357, immediatelyMatchedRequests: 7336, unmatchedRequests: 301, unmatchedPreRequests: 1082 } +6s
  cypress:server:browsers:cri-client closing cri client { closed: false, target: 'ws://127.0.0.1:63456/devtools/browser/54b4426f-e849-478f-8e15-abe2afa65da9' } +2ms
  cypress:cli child event fired { event: 'exit', code: 1, signal: null } +51m
  cypress:cli child event fired { event: 'close', code: 1, signal: null } +1ms`
```

**Comment by illegalnumbers** [comment]
i see something similar when running `start-server-and-test` module

```
WebSocket connection closed
Error: WebSocket connection closed
    at _._handleConnectionClose (<embedded>:2466:579841)
    at A.<anonymous> (<embedded>:2466:577905)
    at Object.onceWrapper (node:events:629:26)
    at A.emit (node:events:514:28)
    at A.emitClose (<embedded>:2466:558129)
    at Socket.W (<embedded>:2466:565440)
    at Socket.emit (node:events:514:28)
    at TCP.<anonymous> (node:net:323:12)
Error: Command failed with exit code 1: npm run cy:run
    at makeError (*MYPATH*/ui/node_modules/start-server-and-test/node_modules/execa/lib/error.js:60:11)
    at handlePromise (*MYPATH*ui/node_modules/start-server-and-test/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
```

**Comment by podlesny-j** [comment]
Having the same issue in GitLab CI with Cypress v13.7.1 running via [cypress/included:13.7.1](https://hub.docker.com/layers/cypress/included/13.7.1/images/sha256-630cc50876c52d7f2cae995b5f62e94a1b1f9d6b30a92237b5986c6a3b96a495?context=explore) image
<img width="739" alt="image" src="https://github.com/cypress-io/cypress/assets/78863563/402c9069-bd11-42fd-a93e-3109501aedb2">

This happens randomly, not in the same spec file in ~5% of all our test runs in CI.

EDIT: sry, I only now noticed this may be a duplicate of https://github.com/cypress-io/cypress/issues/28523 ...?

**Comment by rebel-l** [comment]
> Having the same issue in GitLab CI with Cypress v13.7.1 running via [cypress/included:13.7.1](https://hub.docker.com/layers/cypress/included/13.7.1/images/sha256-630cc50876c52d7f2cae995b5f62e94a1b1f9d6b30a92237b5986c6a3b96a495?context=explore) image <img alt="image" width="739" src="https://private-user-images.githubusercontent.com/78863563/319535226-402c9069-bd11-42fd-a93e-3109501aedb2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NzA5ODIsIm5iZiI6MTcxMjY3MDY4MiwicGF0aCI6Ii83ODg2MzU2My8zMTk1MzUyMjYtNDAyYzkwNjktYmQxMS00MmZkLWE5M2UtMzEwOTUwMWFlZGIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDEzNTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiNDk4NDliNGNlMWYxMjMxZjM0ZjgyN2EyZmFlZDE0MDg0ZDliZjYxMTFhMDliMjlkYzliOWZlZGMzNjBiMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.W_E9dlvAxmJZktCuI1o3lmxdhTiYWSm5Y8VdBy8m_jQ">
> 
> This happens randomly, not in the same spec file in ~5% of all our test runs in CI.
> 
> EDIT: sry, I only now noticed this may be a duplicate of #28523 ...?

yes this issue is a duplicate. see my comment in https://github.com/cypress-io/cypress/issues/28523#issuecomment-1968581673 it's exactly the same error.

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/28523
